pkg/frontend/querymiddleware/split_and_cache.go

```diff
@@ -46,15 +49,27 @@ var (
 
 type splitAndCacheMiddlewareMetrics struct {
 	splitQueriesCount prometheus.Counter
+	notCachableReason *prometheus.CounterVec
 }
 
 func newSplitAndCacheMiddlewareMetrics(reg prometheus.Registerer) *splitAndCacheMiddlewareMetrics {
-	return &splitAndCacheMiddlewareMetrics{
+	m := &splitAndCacheMiddlewareMetrics{
 		splitQueriesCount: promauto.With(reg).NewCounter(prometheus.CounterOpts{
 			Name: "cortex_frontend_split_queries_total",
-			Help: "Total number of underlying query requests after the split by interval is applied",
+			Help: "Total number of underlying query requests after the split by interval is applied.",
 		}),
+		notCachableReason: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Name: "cortex_frontend_not_cachable_reason_total",
```

Comment:
In other places we call this "skipped" (e.g. `cortex_frontend_instant_query_splitting_rewrites_skipped_total`). I think a metric name which is more consistent with similar tracking done in other places is something like `cortex_frontend_query_result_cache_skipped_total`.

_Also, as a general practice, I wouldn't add the "reason" as part of the metric name. It's already a label, and that's enough._

Reply:
Make sense. Thanks!